### ISSUE TITLE
Failing to publish self-contained app to `linux-bionic-arm64`

### ISSUE THREAD

**Issue by eduherminio** [dotnet/sdk#35518]
### Describe the bug

Impossibility to publish a self-contained, single file app to `linux-bionic-arm64` from ubuntu.

Reproduced in [GitHub actions](https://github.com/lynx-chess/Lynx/actions/runs/6233398888/job/16918546076)

```
$ dotnet publish src/Lynx.Cli/Lynx.Cli.csproj -c Release --runtime linux-bionic-arm64 --self-contained /p:Optimized=true /p:DeterministicBuild=true -o artifacts/linux-bionic-arm64
/usr/share/dotnet/sdk/8.0.100-rc.1.23[46](https://github.com/lynx-chess/Lynx/actions/runs/6233398888/job/16918546076#step:7:47)3.5/Microsoft.Common.CurrentVersion.targets(5167,5):
error MSB3030: Could not copy the file "/home/runner/work/Lynx/Lynx/src/Lynx.Cli/obj/Release/net8.0/linux-bionic-arm64/singlefilehost" because it was not found. [/home/runner/work/Lynx/Lynx/src/Lynx.Cli/Lynx.Cli.csproj]
Error: Process completed with exit code 1.
```

, given this in `.csproj`:
```xml
  <PropertyGroup Condition="'$(Optimized.ToLower())'=='true'">
    <Configuration>Release</Configuration>
    <SelfContained>true</SelfContained>
    <PublishSingleFile>true</PublishSingleFile>
    <PublishTrimmed>true</PublishTrimmed>
    <EnableCompressionInSingleFile>true</EnableCompressionInSingleFile>
    <IncludeNativeLibrariesForSelfExtract>true</IncludeNativeLibrariesForSelfExtract>
    <!--In favour of tiered compilation-->
    <PublishReadyToRun>false</PublishReadyToRun>
  </PropertyGroup>
```

### To Reproduce

This commit, where `linux-bionic-arm64` was added to the publish targets https://github.com/lynx-chess/Lynx/commit/f832e05ecfd2c7e6a71414602e78a82b46783eb9

In short, trying to do something like this, cross-compiling from ubuntu-latest (works for `linux-arm64` and any other linux RIDs but this one, see [this run](https://github.com/lynx-chess/Lynx/actions/runs/6233398888/job/16918546076))

```yaml
jobs:
  publish-artifacts:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v4

    - name: Setup .NET
      uses: actions/setup-dotnet@v3
      with:
        dotnet-version: 8.0.x

    - name: Publish ${{ matrix.runtime-identifier }} version
      run: dotnet publish src/Lynx.Cli/Lynx.Cli.csproj -c Release --runtime linux-bionic-arm64 --self-contained /p:Optimized=true /p:DeterministicBuild=true -o artifacts/linux-bionic-arm64
```

### Further technical details
.NET sdk  `8.0.100-rc.1.23463.5`

**Comment by marcpopMSFT** [comment]
@agocke is linux bionic supported for the single file host?

**Comment by eduherminio** [comment]
@agocke Kind ping about this.

**Comment by agocke** [comment]
Thanks for the ping. No, single file was not available for bionic in .NET 8. I will check on the status for .NET 9.

**Comment by TheKingFireS** [comment]
DotNet 9 included SingleFile for `linux-bionic-arm64`. But for some reason, if you publish app with SingleFile and Self-Contain enabled for `linux-bionic-arm64`, it output working binary wihout dotnet libraries (abnormally low size 25mb), if you tried to run that binary, it will complain about missing dotnet libraries. Other (`linux-arm64`/`linux-musl-arm64`) worked fine without issues.

**Comment by agocke** [comment]
This is still unsupported. We don’t yet have an estimate on when coreclr will be available for bionic. I’ll leave this issue open to track providing a better error message.
